Automattic/wp-calypso
# Update plans thank you page design

Fixes https://github.com/Automattic/dotcom-forge/issues/2723

## Proposed Changes

* Update plans checkout thank you page, the `CheckoutThankYou` component design per figma VLoPHWqcewxNKZYjjkDu3O-fi-2414%3A6862
* The styled masterbar was taken (and modified) from the marketplace thank you page. I left the original one in place as its concerns seemed a little different from what we need (at this moment in time anyway).
* Removed leaky css border rule from difm thank you page (I'm somewhat sure this code is not in use anyway and should likely be removed)
* Removed support links

## Todo:
- [ ] We need to flesh out the cta action box with basic details, and fallback details for other purchase types/products.
- [ ] Some of the margins/spacing still need cleaning up
- [ ] Mobile needs a once over
- [ ] Placeholders might need to be replaced with the W logo / check with design?
- [ ] Purchase details need overhaul for every product / purchase
- [ ] Fix unit tests (suspect I broke something)

## Testing Instructions

* Using credits or store sandbox, checkout with a new plan from http://calypso.localhost:3000/plans/
* Compare the figma design
* Repeat for any other thank you pages you might have concerns over
    * Suspect [this list](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/header.jsx#L2-L13) has relatively good coverage
    *  [TitanMail](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/index.tsx#L575), gift purchases, marketplace, and similar `ThankYou` component usage should be unaffected by these changes.

Before
![Screenshot 2023-06-29 at 16-39-09 Thank You — WordPress com](https://github.com/Automattic/wp-calypso/assets/811776/5eea033d-3be8-443a-8b09-8133b16fe987)


After
![Screenshot 2023-06-29 at 16-39-17 Thank You — WordPress com](https://github.com/Automattic/wp-calypso/assets/811776/3cf5d36b-ad34-4863-8e3a-b506c9b823ac)
